app/controllers/sessions_controller.rb

```diff
@@ -28,4 +26,19 @@ def authenticated?(user)
     user && user.authenticate(params[:session][:password])
   end
 
+  def default_login?
+    params[:session]
+  end
+
+  def default_login_info
+    user = User.find_by(username: params[:session][:username])
+    if user && authenticated?(user)
```

Comment:
Is it possible to do this on one line or would that reduce readability?

Reply:
`user if user && authenticated(user)` ?

Reply:
I'm sure we could, but you are right that it may compromise readability. I think we can think about that on a refactor sweep.